src/UnityExtension/Assets/Editor/GitHub.Unity/UI/UserSettingsView.cs

```diff
@@ -100,23 +111,26 @@ private void MaybeUpdateData()
         {
             if (Repository == null)
             {
-                if ((cachedUser == null || String.IsNullOrEmpty(cachedUser.Name)) && GitClient != null)
+                if (!String.IsNullOrEmpty(EntryPoint.Environment.GitExecutablePath))
                 {
-                    var user = new User();
-                    GitClient.GetConfig("user.name", GitConfigSource.User)
-                        .Then((success, value) => user.Name = value).Then(
-                    GitClient.GetConfig("user.email", GitConfigSource.User)
-                        .Then((success, value) => user.Email = value))
-                    .FinallyInUI((success, ex) =>
+                    if ((cachedUser == null || String.IsNullOrEmpty(cachedUser.Name)) && GitClient != null)
                     {
-                        if (success && !String.IsNullOrEmpty(user.Name))
-                        {
-                            cachedUser = user;
-                            userDataHasChanged = true;
-                            Redraw();
-                        }
-                    })
-                    .Start();
+                        GitClient.GetConfigUserAndEmail().FinallyInUI((success, ex, strings) => {
```

Comment:
This is the same code that RepositoryManager is running to load a user, which still feels duplicated. We also shouldn't be calling GitClient directly from the UI, sooo... feels like we should have a User class with static methods that can load itself, it we don't have a Repository/RepositoryManager to do it for us? And then everyone calls that?